test/core/surface/channel_init_test.cc

```diff
@@ -250,18 +253,21 @@ TEST(ChannelInitTest, CanCreateFilterWithCall) {
   b.RegisterFilter<TestFilter1>(GRPC_CLIENT_CHANNEL);
   auto init = b.Build();
   int p = 0;
-  auto segment = init.CreateStackSegment(
-      GRPC_CLIENT_CHANNEL,
-      ChannelArgs().Set("foo", 1).Set("p", ChannelArgs::UnownedPointer(&p)));
-  ASSERT_TRUE(segment.ok()) << segment.status();
-  CallFilters::StackBuilder stack_builder;
-  segment->AddToCallFilterStack(stack_builder);
-  segment = absl::CancelledError();  // force the segment to be destroyed
-  auto stack = stack_builder.Build();
-  {
-    CallFilters call_filters(Arena::MakePooled<ClientMetadata>());
-    call_filters.SetStack(std::move(stack));
-  }
+  InterceptionChainBuilder chain_builder{
+      ChannelArgs().Set("foo", 1).Set("p", ChannelArgs::UnownedPointer(&p))};
+  init.AddToInterceptionChainBuilder(GRPC_CLIENT_CHANNEL, chain_builder);
+  int handled = 0;
```

Comment:
What's the purpose of this counter?  Doesn't look like we're actually checking the value anywhere.

Reply:
missed expectation, added now.